libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/PredictiveCache.kt

```diff
@@ -48,6 +51,7 @@ object PredictiveCache {
     }
 
     fun createMapsController(
+        mapboxMap: Any,
```

Comment:
Do we really need a `MapboxMap` (`Any`) instance here? It seems that we're only interested in the tile variants (`String`) and we're mapping `MapboxMap` / `tileVariant`s internally but not sure if that's needed as we should be only caching once internally (irrespective of the `MapboxMap` instances created) 🤔 

If necessary, does that have any memory implications? Wondering if "storing" `MapboxMap` instances could be causing any leaks or build up in the memory footprint.

Reply:
what if we add 2 map instances with the same tileVariant?
how will we separate caches for 2 maps?
if we have `map` as a key, we can add/remove caches for each map independently (maps may have different lifecycle, different zoom or even show different area, so I think we should have different caches for different map instances). 
If 2 maps show the same area, 2 caches won't duplicate data, I'm I right @mskurydin ?

> If necessary, does that have any memory implications?

no issues as @mskurydin said

> Wondering if "storing" MapboxMap instances could be causing any leaks or build up in the memory footprint

it's a customer responsibility to call add/remove methods to avoid leaks